### PR TITLE
Be more resilient to no unfilteredData. Be better at not trying to render an object as a DataTable cell.

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Be more resilient to no unfilteredData.
Be better at not trying to render an object as a DataTable cell.

#### Where should the reviewer start?

DataFilter.js

#### What testing has been done on this PR?

Added a unit test.

#### How should this be manually tested?

play with stories

#### Do Jest tests follow these best practices?

no change to existing test patterns

#### What are the relevant issues?

#6580 

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DataTable/Cell.js [hunk 1]
```diff
@@ -42,7 +42,16 @@ const Cell = memo(
...
+      if (
+        typeof value === 'string' ||
+        typeof value === 'number' ||
+        isValidElement(value)
+      )
+        content = value;
+      // if (typeof value === 'string' || typeof value === 'number')
+      //   content = value;
+      // else if (typeof value === 'boolean') content = JSON.stringify(value);
+      // else console.log('!!! Cell', typeof value, value);
```
Comment (MikeKingdom) [inline src/js/components/DataTable/Cell.js:54]: remove?

**Review by MikeKingdom — COMMENTED** [review]
Looks good. Just have some commented out code

File: src/js/components/DataTable/Cell.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/Cell.js]: oops, thanks for catching, removed

**Review by jcfilben — APPROVED** [review]
Looks good!

**Pull request merged**